contentcuration/contentcuration/views/admin.py

```diff
@@ -141,6 +142,149 @@ def get_all_users(request):
     return Response(user_serializer.data)
 
 
+class ChannelUserListPagination(PageNumberPagination):
+    page_size = 500
+    page_size_query_param = 'page_size'
+    max_page_size = 500
+
+AND_ = lambda x: 'and_'+x
+OR_ = lambda x: 'or_'+x
+
+@login_required
+@api_view(['GET'])
+@authentication_classes((SessionAuthentication, BasicAuthentication, TokenAuthentication))
+@permission_classes((IsAdminUser,))
+def get_users(request):
+    if not request.user.is_admin:
+        raise SuspiciousOperation("You are not authorized to access this endpoint")
+
+    params = request.GET
+    # return "hey!"
+    paginator = ChannelUserListPagination()
+    user_list = User.objects.prefetch_related('editable_channels').prefetch_related('view_only_channels').distinct()
+    if 'channel_id' in params:
+        channel = Channel.objects.get(pk=params['channel_id'])
+        user_list = user_list.filter(editable_channels__in=[channel])
+    
+    if AND_('is_chef') in params:
+        # import ipdb;ipdb.set_trace()
+        user_list = user_list.annotate(chef_channels=Count('editable_channels__ricecooker_version'))
+        if params.get(AND_('is_chef')) == 'True':
+            user_list = user_list.filter(chef_channels__gt=0)
+        else:
+            user_list = user_list.filter(chef_channels=0)
+
+    query = None
+    for prop in ['is_admin', 'is_active']:
+        if AND_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[AND_(prop)] == 'True' else ~q
+            query = q & query if query else q
+        if OR_(prop) in params:
+            q = Q(**{prop: True})
+            q = q if params[OR_(prop)] == 'True' else ~q
+            query = q | query if query else q
+    
+    if query:
+        user_list = user_list.filter(query)
+
+    if params.get('sort_by') in ['email','fist_name','last_name','channels']:
+        if params.get('ordering') == 'desc':
+            ordering = F(params.get('sort_by')).desc()
+        else:
+            ordering = F(params.get('sort_by')).asc()
+
+        user_list = user_list\
+                    .annotate(channels=Count('editable_channels'))\
+                    .order_by(ordering)
+
+    user_list_page = paginator.paginate_queryset(user_list, request)
+    user_serializer = AdminUserListSerializer(user_list_page, many=True)
+
+    return Response(user_serializer.data)
+
+channel_filters = {
```

Comment:
Quite a few of these are redundant.  For example, `and_is_private=True` could be expressed by `and_is_public=False` .